dashboard/lib/school_info_interstitial_helper.rb

```diff
@@ -24,6 +24,21 @@ def self.show_school_info_interstitial?(user)
     return true
   end
 
+  # Show the school info confirmation dialog when a teacher has either completely
+  # filled out the school info interstitial for a US public, private, or charter school
+  # or confirmed current school over a year ago.
+  def self.show_school_info_confirmation_dialog?(user)
+    return false unless user.teacher?
+
+    school_info = user.school_info
+
+    check_school_type = (school_info.public_school? || school_info.private_school? || school_info.charter_school?) && school_info.school_info_complete?
+
+    check_last_confirmation_date = user.user_school_infos.find(user.school_info_id).last_confirmation_date.to_datetime < 1.year.ago
+
+    check_last_confirmation_date && check_school_type
```

Comment:
Actually, this seems backwards. Shouldn't it be !check_last_confirmation_date? We should show the confirmation dialog if the last confirmation date was more than one year ago.

Reply:
Discussed offline and this is correct as written. `check_last_confirmation_date` checks if the last confirmation date is before today-1year, which is what we want.